### ISSUE TITLE
script to run inference on whole validation dataset

### ISSUE THREAD

**Issue by jettjaniak** [jettjaniak/tinyevals#6]
- we want to run all models on validation dataset
	- [tinystories-v2-clean-tokenized](https://huggingface.co/datasets/jbrinkma/tinystories-v2-clean-tokenized)
	- for now try it on a subset of train dataset
- and collect log probabilities of correct next tokens
  - try it on smallest llamas from [here](https://huggingface.co/collections/jbrinkma/delphi-65abcfefd63812c33e8a6ce1)
- this data will be very useful to search for capabilities at scale
	- without the need to run forward passes each time
- we want to upload this to HF

**Comment by joshuawe** [comment]
- [ ] Which filetype (binary) we want to save the outputs as (Exactly 512 log prob.)
- [ ] We use to log prob. of the CORRECT token not the most likely

**Comment by jettjaniak** [comment]
the dataset now has a working validation split

**Comment by jettjaniak** [comment]
and a note that you can use the function from here to pull the dataset https://github.com/jettjaniak/tinyevals/pull/19

**Comment by transcendingvictor** [comment]
I've done some work on this side.
1) (sequentially) I set up a for loop that basically uses the get_correct_probs function and goes over the whole dataset.
2) (in parallel)  I'm half-way to making the pipeline for the model to process several samples at a time.

Question:
-> shall the values be stored and saved in a specific file, on in a variable within the script that is calculated every time needed?
 
I just commited the changes, I'm working on the validation_script.py file (in brach for issue 6).
